### DISCUSSION TITLE
Nuxt3 - v-slot

### DISCUSSION THREAD

**Discussion by stambekovbera** [nuxt/nuxt#21561] [Questions]
### Environment

npx nuxi info

### Reproduction

Read in additional context

### Describe the bug

Read in additional context

### Additional context

There is such a component (screenshot 1), it recursively renders blocks that are specified in the CMS. A block can have slots, such as the GridBlock block (screenshot 2). The names of all slots are set dynamically, and also in the component that draws the blocks, I also pass the name of the slot dynamically to v-slot, but vue-tsc throws an error (screenshot 3), according to the types slotName is STRING (screenshot 4). In dev mode, everything works fine, rendering goes as it should! The problem is that the project does not build due to this error (Blocks may not have slots! To do this, you can check for the presence of slots in the block object, but everything works fine in dev mode anyway). Help what can be done?

![IntelliJ Snippet](https://github.com/nuxt/nuxt/assets/87119093/dcbcf543-a653-416d-bee8-adfbcac9c932)
![IntelliJ Snippet (1)](https://github.com/nuxt/nuxt/assets/87119093/588e40cf-9c38-4e51-9694-dfd0669b4b30)
![screen](https://github.com/nuxt/nuxt/assets/87119093/7c163c41-d04f-4ff1-a5b6-f35baa2df51b)
![IntelliJ Snippet (2)](https://github.com/nuxt/nuxt/assets/87119093/fba180a5-798f-4eeb-8a0f-c28041c31b04)

**Comment by danielroe** [comment]
You probably need to use `defineSlots` to specify what slots your component accepts.

**Reply by stambekovbera** [reply]
I do not know the names of the slots, they may not be, I only know that if they are, then they will always be of type string
